lib/livebook/hubs/provider.ex

```diff
@@ -60,7 +62,7 @@ defprotocol Livebook.Hubs.Provider do
   """
   @spec create_secret(t(), Secret.t()) ::
           :ok
-          | {:error, Ecto.Changeset.t()}
+          | {:error, field_errors()}
```

Comment:
For secrets (and file systems) the flow is `attrs -> %Changeset{} -> apply changes -> %Secret{} -> Provider.create_secret()`. If `Provider.create_secret` fails, we want to add errors to the original changeset (carrying the form data), rather than returning errors in an empty changeset. The changeset needs the form data for `Phoenix.Component.used_input?` to work properly.

I was considering changing this function to `Provider.create_secret(hub, attrs)`, but I think for secrets and file systems we kinda want the current separation, where the provider already receives the struct. Secrets are create even outside a hub (session secrets), and for file systems we validate the connection before passing to the hub. I did that change for `create_deployment_group` though.